Modules/RatMama.py

```diff
@@ -0,0 +1,178 @@
+    o2_status: bool = result.group("o2") == "OK"
+    lang_code: str = result.group("language_code")
+    nickname: Optional[str] = result.group("nick")
+
+    exist_rescue: Rescue = board.find_by_name(client_name)
+
+    if exist_rescue:
+        # we got a case already!
+        await ctx.reply(f"{client_name} has reconnected! #{exist_rescue.board_index}")
+        # now let's make it more visible if stuff changed
+        diff_response: str = ""
+        if system_name.casefold() != exist_rescue.system.casefold():
+            diff_response += f"System changed! Was {exist_rescue.system}"
+
+        if platform_name.casefold() != str(exist_rescue.platform).replace('Platforms.', '').casefold():
+            diff_response += "Platform changed! "
+
+        if o2_status != exist_rescue.code_red:
+            diff_response += "O2 Status changed" if o2_status else\
+                "O2 Status changed, it is now CODE RED"
+
+        if not diff_response == "":
+            await ctx.reply(diff_response)
+
+    else:
+        # no case for that name, we have to make our own
+        rescue: Rescue = Rescue(client=client_name, system=system_name, irc_nickname=nickname,
+                                code_red=o2_status, lang_id=lang_code)
+        if platform_name.casefold() == "pc":
+            rescue.platform = Platforms.PC
+        elif platform_name.casefold() == "pc":
+            rescue.platform = Platforms.PS
+        elif platform_name.casefold() == "xb":
+            rescue.platform = Platforms.XB
+        else:
+            log.warning(f"Got unknown platform: {platform_name}")
+
+        board.append(rescue, overwrite=False)
+        index = board.find_by_name(client=client_name).board_index
+        await ctx.reply(f"RATSIGNAL - CMDR {client_name} - "
+                        f"Reported System: {system_name} (distance to be implemented) - "
+                        f"Platform: {platform_name} - "
+                        f"O2: {'OK' if o2_status else 'NOT OK'} - "
+                        f"Language: {result.group('full_language')}"
+                        f" (Case #{index}) ({platform_name.upper()}_SIGNAL) "
+                        )
+
+        print(f"RATSIGNAL - CMDR {client_name} - "
+              f"Reported System: {system_name} (distance to be implemented) - "
+              f"Platform: {platform_name} - "
+              f"O2: {'OK' if o2_status else 'NOT OK'} - "
+              f"Language: {result.group('full_language')}"
+              f" (Case #{index}) ({platform_name.upper()}_SIGNAL) "
+              )
+
+
+@rule("ratsignal", case_sensitive=False, full_message=False, pass_match=False, prefixless=True)
+async def handle_selfissued_ratsignal(ctx: Context):
+    message: str = ctx.words_eol[0]
+    message = message.replace("ratsignal", "")  # the ratsignal is nothing we are interested in anymore
+    platform: Platforms
+    cr: bool
+    system: str = ""
+
+    for rescu in board.rescues.values():
```

Comment:
missing e.

Reply:
that is intentional to not get a linter warning regarding shadowing